### PR TITLE
fix: add return on empty user/pass, and do not send superfluous 200 OK

### PR DESCRIPTION
**Pull request by ajcasagrande**
There was a missing return statement in the section that checks if the username or password is empty. Also, warning messages were present that would say superfluous WriteHeader for 200 OK.

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [X] I am not introducing a new dependency (add notes below if you are)
- [N/A] I have added unit tests for the new feature or bug fix (if not, why?)
- [X] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [N/A] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

## Testing Instructions

Build and run this PR using Debug Logging.
Fill out credentials for user/pass for rtsp server
Call Start Streaming API
Expect the logs:
```
level=DEBUG app=device-usb-camera source=driver.go:194 msg="rtsp authentication: username or password is empty"
level=DEBUG app=device-usb-camera source=driver.go:220 msg="rtsp authentication: passwords match"
```

### CODE REVIEW

File: internal/driver/driver.go [hunk 1]
```diff
@@ -172,30 +172,33 @@ func (d *Driver) StartRTSPCredentialServer() {
 	d.rtspAuthServer = nil
 }
 
+// RTSPCredentialsHandler is the http handler for handling rtsp authentication requests. It expects the
+// request to follow the format defined https://github.com/aler9/mediamtx#authentication
```
Comment (presatish) [inline internal/driver/driver.go]:
```suggestion
// request to follow the format defined by https://github.com/aler9/mediamtx#authentication
```

**Review by presatish — DISMISSED** [review]
LGTM

**Comment by vyshali-chitikeshi** [comment]
Validation looks good.
With rtsp user name and password, streaming starts successfully and get below msgs in log
level=DEBUG ts=2023-05-16T00:12:44.735054278Z app=device-usb-camera source=driver.go:194 msg="rtsp authentication: username or password is empty"
level=DEBUG ts=2023-05-16T00:12:44.782961748Z app=device-usb-camera source=driver.go:220 msg="rtsp authentication: passwords match"

Without rtsp user name/password (or empty credentials), start streaming fails and get below msgs in log
level=DEBUG ts=2023-05-16T00:11:16.539368151Z app=device-usb-camera source=driver.go:194 msg="rtsp authentication: username or password is empty"
level=DEBUG ts=2023-05-16T00:11:16.58683335Z app=device-usb-camera source=driver.go:194 msg="rtsp authentication: username or password is empty"

**Review by ChristianDarr-personal — DISMISSED** [review]
LGTM

**Review by presatish — APPROVED** [review]
LGTM

**Pull request merged**